### ISSUE TITLE
ColorMatchTemplate Result is black.

### ISSUE THREAD

**Issue by heeszz** [shimat/opencvsharp#997]
## The Problem

Hello, I have a problem with CvXImgProc.ColorMatchTemplate, maybe someone can help me.

_I installed opencvsharp with NuGet in my C# Project and I am using the Version 4.3.0.20200701 on Windows 10_

Whenever I use ColorMatchTemplate I get an empty (black) result.
Sometimes there is just a little white dot in the black image. But its also on the wrong coordinates.

## Example Code

```
Mat refMat = new Mat(@"bin/img/refMat.PNG", ImreadModes.Color);
Mat tplMat = new Mat(@"bin/img/tplMat.PNG", ImreadModes.Color);
Mat res = new Mat();

CvXImgProc.ColorMatchTemplate(refMat, tplMat, res);
Assert.False(res.Empty());
Assert.Equal(MatType.CV_64FC1, res.Type());

Cv2.ImShow("res", res);
Cv2.WaitKey();
```

## Images:
@"bin/img/refMat.PNG"
![refMat](https://user-images.githubusercontent.com/68499440/87873443-f4229680-c9c1-11ea-8fcd-36cddb35755b.png)

@"bin/img/tplMat.PNG"
![tplMat](https://user-images.githubusercontent.com/68499440/87872616-760ec180-c9ba-11ea-82fd-fdfaada869a6.png)

res
![res](https://user-images.githubusercontent.com/68499440/87873451-fc7ad180-c9c1-11ea-8faa-03fd257d3db0.png)

## Example Code

And here I tried to work with the coordinates, but it doesn't work right when I always get a black picture as a result.
I get the wrong coordinates.

When I edit the threshold to 0.9 or something higher than 0.1 I get no results that's why I put it so low to test it out.
```
double minval, maxval, threshold = 0.01;
OpenCvSharp.Point minloc, maxloc;

res.MinMaxLoc(out minval, out maxval, out minloc, out maxloc);

Mat view = refMat.Clone();
view.Rectangle(maxloc, new OpenCvSharp.Point(maxloc.X + tplMat.Width, maxloc.Y + tplMat.Height), Scalar.Red, 3);

if (maxval >= threshold)
{
   data[1] = maxloc.X.ToString();
   data[2] = maxloc.Y.ToString();
                
   MessageBox.Show(maxval + " " + data[1] + " " + data[2]);

   picturebox3.Image = view.ToBitmap();
   return data;
}
else
{
   return null;
}
                        
```
## Images:

picturebox3.Image = view.ToBitmap();
![picturebox3 Image](https://user-images.githubusercontent.com/68499440/87873495-482d7b00-c9c2-11ea-876b-a892ced34d89.png)

MessageBox.Show(maxval + " " + data[1] + " " + data[2]);
![MessageBox Show](https://user-images.githubusercontent.com/68499440/87873472-292ee900-c9c2-11ea-9f4e-badeca5a9b73.png)

**Comment by heeszz** [comment]
@shimat please can you help me?

**Comment by shimat** [comment]
By normalizing you can see the output image. (It's hard to see, but I can see a faint pattern.)
```c#
Cv2.Normalize(res, res, 0, 1, NormTypes.MinMax);
res.ConvertTo(res, MatType.CV_8UC1, 255);
```

![dst](https://user-images.githubusercontent.com/4434824/88240179-5aa8fc80-ccc1-11ea-83f6-a7096cacc192.png)

I got the same results as you did. I think this may be by design. Your code looks correct. 

I got the right results using the regular MatchTemplate.

```c#
using var src = Image("input.png", ImreadModes.Color);
using var template = Image("template.png", ImreadModes.Color);
using var dst = new Mat();

Cv2.MatchTemplate(src, template, dst, TemplateMatchModes.CCoeffNormed);
Assert.False(dst.Empty());
Assert.Equal(MatType.CV_32FC1, dst.Type());

//Cv2.Normalize(dst, dst, 0, 255, NormTypes.MinMax);

dst.MinMaxLoc(out var minVal, out var maxVal, out var minLoc, out var maxLoc);

using var view = src.Clone();
view.Rectangle(maxLoc, new Point(maxLoc.X + template.Width, maxLoc.Y + template.Height), Scalar.Red, 3);
```

![view](https://user-images.githubusercontent.com/4434824/88240428-09e5d380-ccc2-11ea-834a-4e176c6d87f4.png)

**Comment by heeszz** [comment]
@shimat First of all, thank you so much for taking the time to really help me out.

I thought the problem was in my Code, but maybe colorMatchTemplate isn't the right solution for my problem.

I first tried with MatchTemplate, it gives me the right coordinates and everything is fine. But I have a Problem with normal MatchTemplate. I only want to find a button when it's green but it also finds it when it's gray, so I thought colorMatchTemplate will fix the problem because it's working also with the colors right? But then I got the Problem which we figured out above with the wrong coordinates.

So do you know how to use MatchTemplate but it don't find a match when it's gray?

Again thank you so much!

Green
![Test](https://user-images.githubusercontent.com/68499440/88297177-e3df2400-ccff-11ea-8e73-a0cb49c59b1f.png)

Gray
![Test2](https://user-images.githubusercontent.com/68499440/88297172-e3468d80-ccff-11ea-9769-eced51d81772.png)

**Comment by heeszz** [comment]
I found a fix, I just changed the threshold to 0.98 because the gray RUN has a threshold of 0.97 that's why It always got a match now everything is working fine :)
